### ISSUE TITLE
MSSQL Support

### ISSUE THREAD

**Issue by fmohican** [AuthMe/AuthMeReloaded#1302]
Its possible to provide support for Microsoft SQL Server ?
https://www.microsoft.com/en-us/sql-server/sql-server-2016
Thanks you.

**Comment by Xephi** [comment]
Hi, what's the problem with Microsoft Sql Server? 

Since we already support SQL syntax, is it something into the sql engine that change from all other existing ?
Since we support MySQL, we also support PostgreSQL (without clustering optimisations), InnoDB, etc.. engines, i thinks it actually works well with Microsoft SQL Server, have you test it ?

**Comment by sgdc3** [comment]
@Xephi i think the dialect is different

**Comment by fmohican** [comment]
@Xephi nop i didn't test it, i will test in few moment.

**Comment by fmohican** [comment]
isn't work
https://pastebin.com/phyTP0pM

**Comment by Xephi** [comment]
Huum, ok, @sgdc3 HikariCP alreay handle the Microsoft SQL Server driver

```java
public DataSource getDataSource() {
     HikariConfig config = new HikariConfig();
     config.setDataSourceClassName("com.microsoft.sqlserver.jdbc.SQLServerDataSource");
     config.addDataSourceProperty("serverName", serverName);
     config.addDataSourceProperty("port", port);
     config.addDataSourceProperty("databaseName", databaseName);
     config.addDataSourceProperty("user", user);
     config.addDataSourceProperty("password", password);

     return new HikariDataSource(config);  //pass in HikariConfig to HikariDataSource
}
```
We can add the handle easily i think

**Comment by fmohican** [comment]
would be nice, on hosted server i have to install mysql just for minecraft.... rest of application use MSSQL. xD

**Comment by sgdc3** [comment]
@Xephi, no we have to add the driver, create a MsSql class and change the sql commands

**Comment by sgdc3** [comment]
@fmohican could you try this jar please? https://2079-8662641-gh.circle-artifacts.com/0/tmp/circle-artifacts.icdOyru/AuthMe-5.4-SNAPSHOT.jar

(in the config set the database type as MSSQL before starting)

**Comment by sgdc3** [comment]
try to register/login/changepassword/logout etc
if everything works fine we are almost ready to merge it into the master branch

**Comment by fmohican** [comment]
sure I will test today and I'll get back with reply.

**Comment by fmohican** [comment]
isn't work, i tryied with all port still get same error.
Default port of MSSQL is 1433 or 1434 TCP
https://pastebin.com/mwyXz2Cp

**Comment by sgdc3** [comment]
Whoops my fault xD wrong parameter, fixing it asap

**Comment by sgdc3** [comment]
Fixed: https://2081-8662641-gh.circle-artifacts.com/0/tmp/circle-artifacts.gG5GKtO/AuthMe-5.4-SNAPSHOT.jar

**Comment by fmohican** [comment]
https://pastebin.com/AcNRQ58Z

I trink isn't use login user / password. can u check logs?

**Comment by sgdc3** [comment]
check your db connection credentials (host, port, dbName, user, password) :/

**Comment by fmohican** [comment]
Yes config is right user & password, i just test via php (sqlsvr for php7) script (who connect and try to make a empty table)

**Comment by sgdc3** [comment]
MSSQL requires a complete rewrite of the sql queries, i'll keep this open but not in our todo list atm

**Comment by fmohican** [comment]
okay, Thanks you!

**Comment by sgdc3** [comment]
@fmohican Would PostgreSQL support be useful for you? I think it would be easier to implement and will allow us to start working on an abstraction layer for the sql database implementations ;)

**Comment by fmohican** [comment]
@sgdc3 sorry for late response. Nop for me only MsSql is useful, but I think for others will be useful. it's greate to have more alternative to mysql / mariadb
